Assignment3.java

```diff
@@ -0,0 +1,45 @@
+import java.util.*;
+
+public class Assignment3 {
+    private Set<String> setOfWords = new HashSet<>();
+
+    public List<String> findAllWords(int rowNumber, int columnNumber, char[][] grid, MyDictionary dictionary) {
+        boolean[][] visited = new boolean[rowNumber][columnNumber];
+        String s = "";
+
+        for (int i = 0; i < rowNumber; i++) {
+            for (int j = 0; j < columnNumber; j++) {
+                findWordsStartingWithSpecificChar(grid, i, j, visited, s, dictionary);
+                visited = new boolean[rowNumber][columnNumber];
+            }
+        }
+
+        List<String> words = new ArrayList<>();
+        words.addAll(setOfWords);
+
+        return words;
+    }
+
+    private void findWordsStartingWithSpecificChar(char[][] grid, int row, int column, boolean[][] visited, String s,
+                                                   MyDictionary dictionary) {
+        if (!dictionary.isPrefix(s + grid[row][column])) {
```

Comment:
I have the Dictionary class, but as one of the mentors said in the slack channel and during the Hangout, I just used it for testing, I didn't submit the file.

![screen shot 2017-06-14 at 10 21 25 am](https://user-images.githubusercontent.com/17977547/27120181-47a2cbfc-50eb-11e7-944c-d83810a8aca4.png)